src/pybind/mgr/cephadm/module.py

```diff
@@ -1062,6 +1114,18 @@ def serve(self):
                     r = self._refresh_host_devices(host)
                     if r:
                         failures.append(r)
+
+            if 'CEPHADM_HOST_CHECK_FAILED' in self.health_checks:
+                del self.health_checks['CEPHADM_HOST_CHECK_FAILED']
+            if bad_hosts:
+                self.health_checks['CEPHADM_HOST_CHECK_FAILED'] = {
+                    'severity': 'warning',
+                    'summary': '%d hosts fail cephadm check' % len(bad_hosts),
+                    'count': len(bad_hosts),
+                    'detail': bad_hosts,
+                }
+            self.set_health_checks(self.health_checks)
```

Comment:
this should be indented, and we need it in the previous if too when we delete the item.

or, make a health_changed = False, and set it to True in each of these blocks (below, too!) and do one final set_health_checks call at the bottom if health_changed...